test/controllers/PrivateBetaUsersControllerSpec.scala

```diff
@@ -16,28 +16,258 @@
 
 package controllers
 
-import org.scalatest.freespec.AnyFreeSpec
-import org.scalatest.matchers.must.Matchers
-import play.api.http.Status
-import play.api.libs.json.{JsObject, Json}
+import base.SpecBase
+import models.domain.{Eori, Status, User, UserId}
+import models.requests.NewUserDetails
+import models.{FakeUserIdProvider, UserIdProvider}
+import play.api.http.{Status => HttpStatus}
+import play.api.libs.json._
 import play.api.mvc.Result
 import play.api.test.Helpers._
 import play.api.test._
+import reactivemongo.api.commands.LastError
+import repositories.FakePrivateBetaUserRepository
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class PrivateBetaUsersControllerSpec extends AnyFreeSpec with Matchers {
+class PrivateBetaUsersControllerSpec extends SpecBase {
 
-  private val fakeRequest: FakeRequest[JsObject] = FakeRequest("POST", routes.PrivateBetaUsersController.add().url)
-    .withBody(Json.obj())
+  val unimplementedUserIdProvider = new UserIdProvider {
+    override def generate: UserId = ???
+  }
```

Comment:
These are not implemented, is this intended?

Reply:
Similar as with the FakePrivateBetaUserRepository